python/pyspark/sql/connect/dataframe.py

```diff
@@ -2190,20 +2190,18 @@ def localCheckpoint(
         assert isinstance(checkpointed._plan, plan.CachedRemoteRelation)
         return checkpointed
 
-    if not is_remote_only():
```

Comment:
This is actually intended, e.g., for autocompletetion

Reply:
There are other cases in session.py as well.

Reply:
so we should just skip the test? or remove here and always throw PySparkAttributeError?